src/compile.ts

```diff
@@ -286,12 +306,22 @@ export function compile(
             toExpr(node.right),
           ]);
         } else if (ts.isPrefixUnaryExpression(node)) {
-          if (node.operator !== ts.SyntaxKind.ExclamationToken) {
+          if (
+            ![
+              ts.SyntaxKind.ExclamationToken,
+              ts.SyntaxKind.MinusToken,
+            ].includes(node.operator)
```

Comment:
I think it's simpler to write this logic as:
```ts
if (node.operator !== ts.SyntaxKind.ExclamationToken && node.operator !== ts.SyntaxKind.MinusToken) {
  throw ..
}
```

Also, I don't think `.includes` would narrow the type like the `if` would.